streams/src/test/java/org/apache/kafka/test/ProcessorTopologyTestDriver.java

```diff
@@ -0,0 +1,316 @@
+        ProducerRecord<byte[], byte[]> record = readOutput(topic);
+        if (record == null) return null;
+        K key = keyDeserializer.deserialize(record.topic(), record.key());
+        V value = valueDeserializer.deserialize(record.topic(), record.value());
+        return new ProducerRecord<K, V>(record.topic(), record.partition(), key, value);
+    }
+
+    private Iterable<ConsumerRecord<byte[], byte[]>> records(ConsumerRecord<byte[], byte[]> record) {
+        return Collections.singleton(record);
+    }
+
+    /**
+     * Get the {@link StateStore} with the given name. The name should have been supplied via
+     * {@link #ProcessorTopologyTestDriver(StreamingConfig, TopologyBuilder, String...) this object's constructor}, and is
+     * presumed to be used by a Processor within the topology.
+     * <p>
+     * This is often useful in test cases to pre-populate the store before the test case instructs the topology to
+     * {@link #process(String, byte[], byte[]) process an input message}, and/or to check the store afterward.
+     * 
+     * @param name the name of the store
+     * @return the state store, or null if no store has been registered with the given name
+     * @see #getKeyValueStore(String)
+     */
+    public StateStore getStateStore(String name) {
+        return task.context().getStateStore(name);
+    }
+
+    /**
+     * Get the {@link KeyValueStore} with the given name. The name should have been supplied via
+     * {@link #ProcessorTopologyTestDriver(StreamingConfig, TopologyBuilder, String...) this object's constructor}, and is
+     * presumed to be used by a Processor within the topology.
+     * <p>
+     * This is often useful in test cases to pre-populate the store before the test case instructs the topology to
+     * {@link #process(String, byte[], byte[]) process an input message}, and/or to check the store afterward.
+     * <p>
+     * 
+     * @param name the name of the store
+     * @return the key value store, or null if no {@link KeyValueStore} has been registered with the given name
+     * @see #getStateStore(String)
+     */
+    @SuppressWarnings("unchecked")
+    public <K, V> KeyValueStore<K, V> getKeyValueStore(String name) {
+        StateStore store = getStateStore(name);
+        return store instanceof KeyValueStore ? (KeyValueStore<K, V>) getStateStore(name) : null;
+    }
+
+    /**
+     * Close the driver, its topology, and all processors.
+     */
+    public void close() {
+        task.close();
+    }
+
+    /**
+     * Utility method that creates the {@link MockConsumer} used for restoring state, which should not be done by this
+     * driver object unless this method is overwritten with a functional consumer.
+     * 
+     * @param id the ID of the stream task
+     * @param storeNames the names of the stores that this
+     * @return the mock consumer; never null
+     */
+    protected MockConsumer<byte[], byte[]> createRestoreConsumer(int id, String... storeNames) {
```

Comment:
It's used in the constructor on [line 146](https://github.com/confluentinc/kafka/pull/57/files#diff-a2bf902db43bffc65fe93b819acc218aR146). It's protected in case the subclass wants to override.